### DISCUSSION TITLE
Question About ProjectID Verification and Server Costs

### DISCUSSION THREAD

**Discussion by yushihang** [WalletConnect/walletconnect-monorepo#4637] [Developer Support] [answered]
Hi all,

I have a question regarding the projectID registration and verification process within WalletConnect. 

Specifically, I am curious about the server costs associated with validating projectIDs.

From what I understand:

1. We need to register a projectID on the WalletConnect Cloud dashboard.
2. When sending data over the network, the (decentralized) server validates this projectID to ensure it's a legitimate request.

Given that WalletConnect is an open-source protocol and usage is free, I'm wondering who covers the costs for the servers that handle projectID verification. 

Are these costs covered by WalletConnect, or do you have a different business model in place (e.g., enterprise fees, sponsorships, etc.) to manage these expenses?

Additionally, if there are any official documents or resources that explain this aspect in detail, I would appreciate it if you could point me in that direction.

Thank you for your time and assistance.

**Comment by glitch-txs** [comment, marked as answer]
Hello, the server costs are covered by WalletConnect and the Relay is currently cetralized. This will change while moving towards decentralizing the Network. You can read more about it [here](https://walletconnect.com/blog/the-next-phase-of-the-walletconnect-protocol).

**Reply by yushihang** [reply]
Thanks a lot for your answer

**Reply by yushihang** [reply]
@glitch-txs 

I have two more questions I'd like to consult with you about. Thank you for your time and patience.

1. From these documents, such as: <https://specs.walletconnect.com/2.0/glossary/relay> (By default, the clients will use a proxy server connected to the `Waku` network …), it still mentions that WalletConnect uses the Waku-related network.

Does this mean that the centralized Relay network maintained by WalletConnect is still similar to the Waku architecture, or is it just because the documents have not been updated?

2. I see a lot of discussions in the discuss list about "whether WalletConnect's Relay network supports self-deployment/hosting", and the responses indicate that the next step is mainly to modify the relay to be decentralized rather than self-hosting.

    I would like to confirm, as of now, does WalletConnect have plans to support self-hosting for the Relay network?

Thanks a lot~

**Reply by glitch-txs** [reply]
WalletConnect doesn't use Waku, the link you shared needs to be updated. At first, we tried to use Waku but we couldn't resolve the latency issue so we built our own Relay.

Self-hosting is still a way of centralization, you're just changing the owner.
 What we are currently working on is a single Network that can be operated and maintained by different nodes. Anyone will be able to run a node, but the WalletConnect network will be only one

**Reply by yushihang** [reply]
@glitch-txs  Thank you for your patient and detailed explanation.👍
 This has deepened my understanding of WalletConnect.

> Self-hosting is still a way of centralization, you're just changing the owner

Yes, you are right.

We are considering using WalletConnect in our product, but we may face some internal security audit requirements. Therefore, if WalletConnect supports self-hosting, it would be easier for us to pass the security audit.😂
(Because under the assumption of self-hosting, the relay network nodes are deployed and maintained by us.)

**Reply by glitch-txs** [reply]
We do run audits internally. Feel free to contact us at security@walletconnect.com for security related questions

Also: https://medium.com/walletconnect/walletconnect-v2-0s-independent-security-audit-by-trail-of-bits-e8e59d11e517

**Reply by yushihang** [reply]
> We do run audits internally. Feel free to contact us at [security@walletconnect.com](mailto:security@walletconnect.com) for security related questions
> 
> Also: https://medium.com/walletconnect/walletconnect-v2-0s-independent-security-audit-by-trail-of-bits-e8e59d11e517

This is very helpful. Thank you again!
